src/rgw/rgw_admin.cc

```diff
@@ -4811,6 +4828,50 @@ int main(int argc, const char **argv)
     }
   }
 
+  if (opt_cmd == OPT_BUCKETS_LIMIT_CHECK) {
+    void *handle;
+    std::list<std::string> user_ids;
+    metadata_key = "user";
+    int max = 1000;
+    int ret = store->meta_mgr->list_keys_init(metadata_key, &handle);
```

Comment:
this call isn't needed in the `if (! user_id.empty()) {` case (and will leak memory because `list_keys_complete()` is only called in the `} else {` block)